Snowflake.Client/SnowflakeClient.cs

```diff
@@ -117,14 +116,13 @@ public async Task<bool> RenewSessionAsync()
             if (_session == null)
                 throw new SnowflakeException("Session is not itialized yet.");
 
-            var renewSessionRequest = _requestBuilder.BuildRenewSessionRequest();
+            var renewSessionRequest = _requestBuilder.BuildRenewSessionRequest(_session);
             var response = await _restClient.SendAsync<RenewSessionResponse>(renewSessionRequest).ConfigureAwait(false);
 
             if (!response.Success)
                 throw new SnowflakeException($"Renew session failed. Message: {response.Message}", response.Code);
 
-            _session.Renew(response.Data);
```

Comment:
Might be nice to keep this method but have it call the new ctor intenrally. Then this becomes:
```csharp
_session = _session.Renew(response.Data);
```

Reply:
this means that `SnowflakeSession` remains mutable though.